### PR TITLE
Kibana chart should have service account

### LINKED ISSUE #7059: Kibana chart should have service account
**Which chart**:
The name (and version) of the affected chart.

Kibana latest

**Is your feature request related to a problem? Please describe.**

In order to give service accounts the fine-grained access control that they need and not just give the default service account permissions, a service account should be defined for a deployment's pods.  In the case of kibana, in some environments it needs the anyuid security context constraint.  We may not want to provide this to default, plus doing so may be blocked by auditing + infosec.

**Describe the solution you'd like**

Like the elasticsearch chart, for the kibana chart there should be a `serviceAccount` section with two key-values: name (string), and create (bool).  If not specified, the default service account may be used.

**Describe alternatives you've considered**

I think adding the above should be straightforward enough.  We're using the kibana subchart with `kibanaEnabled: true` from the elasticsearch chart, so things ShouldJustWork.  One alternative is not to use the kibana subchart and go with the official kibana chart from elasticsearch.

**Additional context**

We're installing on openshift and aiding openshift customers to do so.  They have infosec requirements that we do not control.

### PR DESCRIPTION
**Pull request by jeff-hammel-leapyear**
**Description of the change**

This allows the specification of a service account for Kibana.  See also https://github.com/bitnami/charts/issues/7059

**Benefits**

You should now be able to specify security contexts to a particular service account, not just defaults.

**Possible drawbacks**

None that I can think of.

**Applicable issues**

  - fixes #7059 

**Additional information**

**Checklist** 

- [ ] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [ ] Variables are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

**Comment by jeff-hammel-leapyear** [comment]
I'm not going to have a chance to look at this soon.  Could someone please comment on what needs to be done to get this PR in a state?  Alternately, if someone could shepherd this in, I would be grateful.

File: bitnami/kibana/templates/serviceaccount.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{{- if.Values.serviceAccount.create }}
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name:  {{ required .Values.serviceAccount.name }}
+{{- end }}
```
Comment (miguelaeh) [inline bitnami/kibana/templates/serviceaccount.yaml:6]: Could you please base on the following service account in order to standardize it please? https://github.com/bitnami/charts/blob/master/bitnami/redis/templates/serviceaccount.yaml

File: bitnami/kibana/templates/deployment.yaml [hunk 2]
```diff
@@ -73,6 +73,7 @@ spec:
           securityContext:
             runAsUser: {{ .Values.securityContext.runAsUser }}
           {{- end }}
+          serviceAccountName: {{ .Values.serviceAccount.name }}
```
Comment (miguelaeh) [inline bitnami/kibana/templates/deployment.yaml:76]: We use a template in order to render the proper serviceAccount name, you can base on the following: https://github.com/bitnami/charts/blob/c10c7f4015d1ed0bf52ed10f9cd0cc6fb7d94910/bitnami/redis/templates/_helpers.tpl#L141

**Review by miguelaeh — CHANGES_REQUESTED** [review]
Hi @jeff-hammel-leapyear ,
Thank you very much for the change.
Please take a look at my comments, also don't forget to bump the chart version. There is a checklist on the PR comment that you should complete.

**Comment by james-pollard-leapyear** [comment]
@miguelaeh Hi, Jeff's teammate here. He's on vacation for a while and I imagine you can understand that this change isn't a priority for us. Please consider his change a PoC and take lead on shepherding this change through.

Thanks for getting back to us!

**Comment by Mauraza** [comment]
Hi @james-pollard-leapyear,

I just create a new PR with the change.

**Comment by james-pollard-leapyear** [comment]
> I just create a new PR with the change.

Excellent - thank you! feel free to close out our PR here

**Comment by juan131** [comment]
Implemented at https://github.com/bitnami/charts/pull/7211

**Comment by james-pollard-leapyear** [comment]
Thanks a bunch!

**Pull request closed without merging**